addons/portal/controllers/mail.py

```diff
@@ -55,9 +55,10 @@ def portal_chatter_init(self, thread_model, thread_id, **kwargs):
                 can_react = has_react_access and portal_partner
                 partner = portal_partner or partner
             store.add(thread, {"can_react": bool(can_react)}, as_thread=True)
-        store.add({"self": Store.one(partner, fields=["active", "avatar_128", "name", "user"])})
-        if request.env.user.has_group("website.group_website_restricted_editor"):
-            store.add(partner, {"is_user_publisher": True})
+        if partner:
+            store.add({"self": Store.one(partner, fields=["active", "avatar_128", "name", "user"])})
```

Comment:
In non-portal context, guest should take priority over portal partner.

If the live chat is started on a portal page with a portal partner, and then the user moves to another non-portal chatter page, the portal partner will no longer be available.